cogs/wordfilter/wordfilter.py

```diff
@@ -115,6 +124,9 @@ async def removeFilter(self, ctx, word: str):
         else:
             filters.remove(word)
             await self.config.guild(ctx.guild).get_attr(KEY_FILTERS).set(filters)
+            filterStats = await self.config.guild(ctx.guild).get_attr(KEY_USAGE_STATS)()
+            del filterStats[word]
+            await self.config.guild(ctx.guild).get_attr(KEY_USAGE_STATS).set(filterStats)
```

Comment:
Either guard the `del` line by doing an existence check for `word` in `filterStats`,
```suggestion
            if word in filterStats:
                del filterStats[word]
                await self.config.guild(ctx.guild).get_attr(KEY_USAGE_STATS).set(filterStats)
```
or register the word right away when it is added using `[p]wordfilter regex add`.

I am not sure how events are triggered, but I did not see `KeyError` when I was testing on my local machine. It appeared that `checkWords()` was invoked right after `addFilter()` (a.k.a. `[p]wordfilter regex add`) had done its job, and so every new word got into `filterStats`, but speaking from the code alone, this `del` line would raise `KeyError` if `word` were not in `filterStats`.